datasets/_global/gem_energy_ownership/crawler.py

```diff
@@ -96,16 +90,19 @@ def crawl_company(context: Context, row: Dict[str, str], skipped: Set[str]):
     if homepage:
         entity.add("website", split_urls(homepage))
     if schema != "Person":
-        entity.add("permId", perm_id)
-        # find a way to remap invalid ones
-        entity.add("ogrnCode", ru_id)
+        entity.add_cast("Company", "permId", perm_id)
+        # Check if ru_id is a valid OGRN
+        if ru_id and rigour.ids.OGRN.is_valid(ru_id):
+            entity.add("ogrnCode", ru_id)
+        else:  # Remap invalid ones
+            entity.add("registrationNumber", ru_id)
         entity.add("registrationNumber", br_id)
         entity.add("registrationNumber", uk_id)
         entity.add("registrationNumber", in_id)
         entity.add("registrationNumber", us_eia_id)
         entity.add("registrationNumber", sp_cap)
         if schema != "PublicBody":
-            entity.add("cikCode", us_sec_id)
+            entity.add_cast("Company", "cikCode", us_sec_id)
         else:  # PublicBody
             entity.add("registrationNumber", us_sec_id)
```

Comment:
not required, but as a tiny tidy, if the first clause is `if entity.schema.is_a("PublicBody")`, it's more obvious that the second clause is not a public body, so the comment isn't needed. No double negative in your head :)
the is_a is also more future-proof